### PR TITLE
remove check for downloadrequest

### PR DESCRIPTION
**Pull request by cleverhu**
Thank you for contributing to Velero!

# Please add a summary of your change
remove check for downloadrequest
# Does your change fix a particular issue?

Fixes https://github.com/vmware-tanzu/velero/issues/3126

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by reasonerjt — APPROVED** [review]
lgtm, thanks

File: pkg/cmd/util/downloadrequest/downloadrequest.go [hunk 1]
```diff
@@ -72,12 +72,6 @@ func Stream(ctx context.Context, kbClient kbclient.Client, namespace, name strin
 			return
 		}
 
-		// TODO: once the minimum supported Kubernetes version is v1.9.0, remove the following check.
-		// See http://issue.k8s.io/51046 for details.
-		if updated.Name != created.Name {
-			return
-		}
-
```
Comment (sseago) [inline pkg/cmd/util/downloadrequest/downloadrequest.go:80]: I'm not quite sure I understand the relationship between this check and the issue referenced. The issue seems to be about adding fieldSelector support for CRDs, but where is Velero's use of field selectors dependent on the above check?

Comment (cleverhu) [reply pkg/cmd/util/downloadrequest/downloadrequest.go:80]: According to the pr https://github.com/vmware-tanzu/velero/pull/3004#discussion_r533720827,
```
Pretty sure we can actually remove these checks now. However, let's not in this PR :)
```
There is a need to clean up the code.

Comment (cleverhu) [reply pkg/cmd/util/downloadrequest/downloadrequest.go:80]: This is a forgotten child issue.

Comment (sseago) [reply pkg/cmd/util/downloadrequest/downloadrequest.go:80]: I saw that. But I'm still unclear how these things are related. I don't doubt that this is likely unnecessary now, but I'm just not seeing how the change referenced there relates to whether or not we need this. How is this check related to the field selectors changed referenced in http://issue.k8s.io/51046

**Pull request closed without merging**